File: 2915-count-of-interesting-subarrays/2915-count-of-interesting-subarrays.ts
function countInterestingSubarrays(nums: number[], modulo: number, k: number): number {
    const hashmap = new Map<number, number>()
    hashmap.set(0, 1)
    let count = 0;
    let sum = 0;

    for (let value of nums) {
        sum += value % modulo === k ? 1 : 0;
        count += (hashmap.get((sum + modulo - k) % modulo) || 0);
        hashmap.set(sum % modulo, (hashmap.get(sum % modulo) || 0) + 1);
    }
    return count;
};

File: 2915-count-of-interesting-subarrays/2915-count-of-interesting-subarrays.py
class Solution:
    def countInterestingSubarrays(self, nums: List[int], modulo: int, k: int) -> int:
        hashmap = {0: 1}
        count = 0
        s = 0

        for num in nums:
            s += 1 if num % modulo == k else 0
            sMod = s % modulo

            a = (sMod - k + modulo) % modulo
            count += hashmap.get(a, 0)
            hashmap[sMod] = hashmap.get(sMod, 0) + 1
        
        return count